---
title: "Grene_DataAnalysis_Final"
author: "liuyaping"
date: "2025-08-05"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      dev = "png")
```

```{r library,results='hide'}
#library R packages.
library(vegan)
library(ggplot2)
library(ggpmisc)
library(here)
library(tidyverse)
library(dplyr)#rename
library(ggsignif)
library(cowplot)
library(rfPermute)
library(tidyr)
library(gdm)
library(geosphere)
library(plspm) # SEM
library(rstatix)
library(reshape)#melt
library(kableExtra)#output of table
library(linkET)#mantel
library(phyloseq)
library(ggpubr) #stat_compare_means
```

```{r path}
#Use the "here" function to use relative path.
here::here("dataanalysis")
```

```{r get data,base data, microbial data and GDM output data}
# get base data
data_path <- here("dataanalysis/00data/rawdata","ASV-sample.csv")
data_base <- read.csv(data_path)

data_path_38 <- here("dataanalysis/00data","sem_for_alpha2_38.txt")
data_base_38 <- read.csv(data_path_38, header = TRUE, sep = "\t", row.names = 1)

data_path_190 <- here("dataanalysis/00data","sem_beta190.txt")
data_base_190 <- read.csv(data_path_190, header = TRUE, sep = "\t", row.names = 1)

# get microbial data
## bacteria
otu_tab_bac_path <- here("dataanalysis/00data/rawdata","Bac-count.csv")
otu_tab_bac <- read.csv(otu_tab_bac_path,fileEncoding = "UTF-8-BOM")

tax_tab_bac_path <- here("dataanalysis/00data/rawdata","Bac-taxonomy.csv")
tax_tab_bac <- read.csv(tax_tab_bac_path,fileEncoding = "UTF-8-BOM")

## fungi
otu_tab_fung_path <- here("dataanalysis/00data/rawdata","fungi-count.csv")
otu_tab_fung <- read.csv(otu_tab_fung_path,fileEncoding = "UTF-8-BOM")

tax_tab_fung_path <- here("dataanalysis/00data/rawdata","fungi-taxonomy.csv")
tax_tab_fung <- read.csv(tax_tab_fung_path,fileEncoding = "UTF-8-BOM")

# get gdm output
gdm_bac_path <- here("dataanalysis/00data","GRENE_bac_GDM_ispline.txt")
gdm_bac <- read.csv(gdm_bac_path, header = TRUE, sep = "\t", row.names = 1)

gdm_fung_path <- here("dataanalysis/00data","GRENE_fung_GDM_ispline.txt")
gdm_fung <- read.csv(gdm_fung_path, header = TRUE, sep = "\t", row.names = 1)

# get function data
KO_N_path <-  here("dataanalysis/00data","KO_nitrogen_summary_38.txt")
KO_N <- read.csv(KO_N_path, header = TRUE, sep = "\t", row.names = 1)

KO_path <- here("dataanalysis/00data","MAPLE_KO_summary_for_R_38.txt")
KO <- read.csv(KO_path, header = TRUE, sep = "\t")
KO_table <- KO %>%
  group_by(KO) %>%  
  summarise(across(everything(), sum))
KO_table <- data.frame(KO_table)
rownames(KO_table) <- KO_table$KO
KO_table <- KO_table[,-1]

```

```{r import data into phyloseq}
# Import into phyloseq
## Bacteria
row.names(otu_tab_bac) <- otu_tab_bac$OTU
otu_tab_bac <- otu_tab_bac %>% select(-OTU)
row.names(tax_tab_bac) <- tax_tab_bac$OTU
tax_tab_bac <- tax_tab_bac %>% select (-OTU) 
row.names(data_base) <- data_base$sample
samples <- data_base %>% select (-sample)
otu_tab_bac <- as.matrix(otu_tab_bac)  # Transform into matrixes otu and tax tables
tax_tab_bac <- as.matrix(tax_tab_bac)
otu_tab_bac = otu_table(otu_tab_bac, taxa_are_rows = TRUE) # Transform into phyloseq readable format
tax_tab_bac = tax_table(tax_tab_bac)
samples = sample_data(data_base)
physeq_bac <- phyloseq(otu_tab_bac, tax_tab_bac, samples)

## Fungi
row.names(otu_tab_fung) <- otu_tab_fung$OTU
otu_tab_fung <- otu_tab_fung %>% select(-OTU)
row.names(tax_tab_fung) <- tax_tab_fung$OTU
tax_tab_fung <- tax_tab_fung %>% select (-OTU) 
row.names(data_base) <- data_base$sample
samples <- data_base %>% select (-sample)
otu_tab_fung <- as.matrix(otu_tab_fung)    # Transform into matrixes otu and tax tables
tax_tab_fung <- as.matrix(tax_tab_fung)
otu_tab_fung = otu_table(otu_tab_fung, taxa_are_rows = TRUE) # Transform into phyloseq readable format
tax_tab_fung = tax_table(tax_tab_fung)
samples = sample_data(data_base)
physeq_fungi <- phyloseq(otu_tab_fung, tax_tab_fung, samples)

## N cycling
physeq_N <- phyloseq(
  otu_table(KO_N, taxa_are_rows = TRUE),
  sample_data(data_base_38)
)

## KO
## N cycling
physeq_KO <- phyloseq(
  otu_table(KO_table, taxa_are_rows = TRUE),
  sample_data(data_base_38)
)

```


```{r output path}
Figure_path <- "C:/Users/alip/Dropbox/Yaping/GRENE/paper/Revise_submit_20250728/Revise_submit_20250728/RScriptsandFigures/"
```


>**Figure1: Broad-scale factors => Soil properties**

```{r Random forest}
## braod-scale factors
# WRB = "soil patent material"
# Dominant_tree_type = "vegetation"
# Dominant_tree_species = "tree species"
# MAP and MAT = "climate"

## soil properties
# pH.H2O. = "soil habitat pH"
# Total.C = "soil C content"
# Total.N = "soil N content"
# CN = "soil C-to-N"
# FH_C = "litter C content"
# FH_N = "litter N content"
# FH_CN = "litter C-to-N"

set.seed(123)
## Random forest for pH.H2O.
RF_pH <- data_base[,c("pH.H2O.","WRB","Dominant_tree_type","MAP", "MAT")] # this dataset includes only the variables of interest
rfp_pH <- rfPermute(pH.H2O.~., data = RF_pH, importance = TRUE, ntree = 500, nrep = 1000, num.cores = 1)
importance_pH <- data.frame(Variable = rownames(importance(rfp_pH, scale = TRUE)),
                            importance(rfp_pH, scale = TRUE), 
                            heck.names = FALSE)

## Add a new column for asterisk labels based on the p-value condition
importance_pH$label <- ifelse(importance_pH$X.IncMSE.pval < 0.05, "*", "")
print(importance_pH)

## Extract the last value of rfp_pH$rf$rsq, which corresponds to the overall % Var explained
var_explained_pH <- tail(rfp_pH$rf$rsq, 1) * 100
print(var_explained_pH)

## Plot
RF_pH_plot <-
  ggplot(importance_pH, aes(x = reorder(Variable, X.IncMSE), y = X.IncMSE)) +
  geom_bar(stat = "identity", fill = "gray", alpha = 0.8) +
  geom_text(aes(label = label, y = X.IncMSE + 2)) + # add P-value label
  coord_flip() + # Flip coordinates for horizontal bars
  labs(x = "", y = "Increase in MSE (%)", 
       subtitle = paste("Soil pH\nVariation explained:", round(var_explained_pH, 2), "%")) + # add var explained
  theme_classic() +
  theme(axis.title =  element_text(size = 12),
        axis.text = element_text(size = 12),
        plot.subtitle = element_text(size = 12)) + 
  scale_x_discrete(labels = function(x) sapply(x, function(el) {
    if (el == "WRB") return("Soil type")
    else if (el == "Dominant_tree_type") return("Vegetation")
    else return(el)
  })) 
print(RF_pH_plot)

ggsave(paste0(Figure_path,"Figure1B_RF_pH_plot.pdf"),RF_pH_plot,width = 3,height = 4)

## Random forest for soil C content
RF_TC <- data_base[,c("Total.C","WRB","Dominant_tree_type","MAP", "MAT")] # this dataset includes only the variables of interest
rfp_TC <- rfPermute(Total.C~., data = RF_TC, importance = TRUE, ntree = 500, nrep = 1000, num.cores = 1)
importance_TC <- data.frame(Variable = rownames(importance(rfp_TC, scale = TRUE)),
                            importance(rfp_TC, scale = TRUE), 
                            heck.names = FALSE)

## Add a new column for asterisk labels based on the p-value condition
importance_TC$label <- ifelse(importance_TC$X.IncMSE.pval < 0.05, "*", "")
print(importance_TC)

## Extract the last value of rfp_pH$rf$rsq, which corresponds to the overall % Var explained
var_explained_TC <- tail(rfp_TC$rf$rsq, 1) * 100
print(var_explained_TC)

## Plot
RF_TC_plot <-
  ggplot(importance_TC, aes(x = reorder(Variable, X.IncMSE), y = X.IncMSE)) +
  geom_bar(stat = "identity", fill = "gray", alpha = 0.8) +
  geom_text(aes(label = label, y = X.IncMSE + 2)) + # add P-value label
  coord_flip() + # Flip coordinates for horizontal bars
  labs(x = "", y = "Increase in MSE (%)", 
       subtitle = paste("Soil C content/nVariation explained:", round(var_explained_TC, 2), "%")) + # add var explained
  theme_classic() +
  theme(axis.title =  element_text(size = 12),
        axis.text = element_text(size = 12),
        plot.subtitle = element_text(size = 12)) + 
  scale_x_discrete(labels = function(x) sapply(x, function(el) {
    if (el == "WRB") return("Soil type")
    else if (el == "Dominant_tree_type") return("Vegetation")
    else return(el)
  })) 
print(RF_TC_plot)

ggsave(paste0(Figure_path,"Figure1D_RF_TC_plot.pdf"),RF_TC_plot,width = 3,height = 4)

## Random forest for soil N content
RF_TN <- data_base[,c("Total.N","WRB","Dominant_tree_type","MAP", "MAT")] # this dataset includes only the variables of interest
rfp_TN <- rfPermute(Total.N~., data = RF_TN, importance = TRUE, ntree = 500, nrep = 1000, num.cores = 1)
importance_TN <- data.frame(Variable = rownames(importance(rfp_TN, scale = TRUE)),
                            importance(rfp_TN, scale = TRUE), 
                            heck.names = FALSE)
## Add a new column for asterisk labels based on the p-value condition
importance_TN$label <- ifelse(importance_TN$X.IncMSE.pval < 0.05, "*", "")
print(importance_TN)

## Extract the last value of rfp_pH$rf$rsq, which corresponds to the overall % Var explained
var_explained_TN <- tail(rfp_TN$rf$rsq, 1) * 100
print(var_explained_TN)

## Plot
RF_TN_plot <-
  ggplot(importance_TN, aes(x = reorder(Variable, X.IncMSE), y = X.IncMSE)) +
  geom_bar(stat = "identity", fill = "gray", alpha = 0.8) +
  geom_text(aes(label = label, y = X.IncMSE + 2)) + # add P-value label
  coord_flip() + # Flip coordinates for horizontal bars
  labs(x = "", y = "Increase in MSE (%)", 
       subtitle = paste("Soil N content\nVariation explained:", round(var_explained_TN, 2), "%")) + # add var explained
  theme_classic() +
  theme(axis.title =  element_text(size = 12),
        axis.text = element_text(size = 12),
        plot.subtitle = element_text(size = 12)) + 
  scale_x_discrete(labels = function(x) sapply(x, function(el) {
    if (el == "WRB") return("Soil type")
    else if (el == "Dominant_tree_type") return("Vegetation")
    else return(el)
  })) 
print(RF_TN_plot)

ggsave(paste0(Figure_path,"Figure1F_RF_TN_plot.pdf"),RF_TN_plot,width = 3,height = 4)

```


```{r Boxplot}
# Boxplot of pH.H2O. vs Vegetation
pHvsVeg_plot <- 
  ggplot(data = na.omit(data_base[c('Dominant_tree_type', 'pH.H2O.')]), aes(x = Dominant_tree_type, y = pH.H2O.)) +
  geom_boxplot(aes(fill = Dominant_tree_type)) + # Fill boxes based on Dominant_tree_type +
  geom_jitter(color = "black", width = 0.2, alpha = 0.5) + # Add jitter with specified color and transparency
  scale_fill_manual(values = c("Broadleaf" = "#52854C", "Conifer" = "#E7B800")) + # Custom colors for box fill
  theme_classic() + 
  theme(axis.title = element_text(size = 12), 
        axis.text = element_text(size = 12, angle = 30, hjust = 1),
        legend.position = "none") +
  labs(x = "Vegetation", y = "Soil habitat pH") + 
  scale_y_continuous(limits = c(3, 8.5)) + # Adjust the y-axis scale
  stat_compare_means(method = "kruskal.test", label.y = 8.2) # Perform and add statistical comparison
plot(pHvsVeg_plot)

ggsave(paste0(Figure_path,"Figure1C_Box_pH_plot.pdf"),pHvsVeg_plot,width = 3,height = 4)

# Boxplot of Total.C vs Soil Parent Material
TCvsSoil_plot <-   
  ggplot(data = na.omit(data_base[c('WRB', 'Total.C')]), aes(x = WRB, y = Total.C)) +
  geom_boxplot(aes(fill = WRB)) + 
  geom_jitter(color = "black", width = 0.2, alpha = 0.5) + 
  scale_fill_manual(values = c("Acrisols" = "gray", "Andosols" = "brown","Cambisols" = "#1f77b4", "Regosols" = "gray")) + 
  theme_classic() + 
  theme(axis.title = element_text(size = 12), 
        axis.text = element_text(size = 12, angle = 30, hjust = 1),
        legend.position = "none") +
  labs(x = "Soil type", y = "Soil C content (%)") + 
  scale_y_continuous(limits = c(0, 32)) + # Adjust the y-axis scale
  stat_compare_means(comparisons = list( c("Andosols", "Cambisols"), label.y = 23)) +
  stat_compare_means(method = "kruskal.test", label.y = 30) # Perform and add statistical comparison
plot(TCvsSoil_plot)

ggsave(paste0(Figure_path,"Figure1E_Box_TC_plot.pdf"),TCvsSoil_plot,width = 3,height = 4)

# Boxplot of Total.N vs Soil Parent Material
TNvsSoil_plot <-   
  ggplot(data = na.omit(data_base[c('WRB', 'Total.N')]), aes(x = WRB, y = Total.N)) +
  geom_boxplot(aes(fill = WRB)) + 
  geom_jitter(color = "black", width = 0.2, alpha = 0.5) + 
  scale_fill_manual(values = c("Acrisols" = "gray", "Andosols" = "brown","Cambisols" = "#1f77b4", "Regosols" = "gray")) + 
  theme_classic() + 
  theme(axis.title = element_text(size = 12), 
        axis.text = element_text(size = 12, angle = 30, hjust = 1),
        legend.position = "none") +
  labs(x = "Soil type", y = "Soil N content (%)") + 
  scale_y_continuous(limits = c(0, 1.8)) + # Adjust the y-axis scale
  stat_compare_means(comparisons = list( c("Andosols", "Cambisols"))) +
  stat_compare_means(method = "kruskal.test", label.y = 1.7) # Perform and add statistical comparison
plot(TNvsSoil_plot)

ggsave(paste0(Figure_path,"Figure1G_Box_TN_plot.pdf"),TNvsSoil_plot,width = 3,height = 4)

```

```{r save sub-figures of Figure1}
# Save
#ggsave(paste0(figure1_path, "/RF_pH_plot.png"), RF_pH_plot, width = 3.5, height = 4, dpi = 300)
#ggsave(paste0(figure1_path, "/RF_TC_plot.png"), RF_TC_plot, width = 3.5, height = 4, dpi = 300)
#ggsave(paste0(figure1_path, "/RF_TN_plot.png"), RF_TN_plot, width = 3.5, height = 4, dpi = 300)
#ggsave(paste0(figure1_path, "/pHvsVeg_plot.png"), pHvsVeg_plot, width = 3.5, height = 4, dpi = 300)
#ggsave(paste0(figure1_path, "/TCvsSoil_plot.png"), TCvsSoil_plot, width = 3.5, height = 4, dpi = 300)
#ggsave(paste0(figure1_path, "/TNvsSoil_plot.png"), TNvsSoil_plot, width = 3.5, height = 4, dpi = 300)
```


>**Figure2: Soil properties => Microbial abundance and richness**

```{r calculate alpha and beta diverstiy}
# Rarefy OTU table
## Bacteria
set.seed(99)
physeq.r.bac <- rarefy_even_depth(physeq_bac, sample.size =min(sample_sums(physeq_bac)),rngseed = FALSE, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
physeq.r.bac
#write.table(otu_table(physeq.r.bac), file = here("dataanalysis/00data/rawdata","bacteria-count-190.txt"),sep = "\t",row.names = T,col.names = NA,quote = F)
#write.table(tax_table(physeq.r.bac), file = here("dataanalysis/00data/rawdata","bacteria-taxonomy-190.txt"),sep = "\t",row.names = T,col.names = NA,quote = F)
## Fungi
set.seed(99)
physeq.r.fungi <- rarefy_even_depth(physeq_fungi, sample.size =min(sample_sums(physeq_fungi)),rngseed = FALSE, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
physeq.r.fungi
#write.table(otu_table(physeq.r.fungi), file = here("dataanalysis/00data/rawdata","fungi-count-190.txt"),sep = "\t",row.names = T,col.names = NA,quote = F)
#write.table(tax_table(physeq.r.fungi), file = here("dataanalysis/00data/rawdata","fungi-taxonomy-190.txt"),sep = "\t",row.names = T,col.names = NA,quote = F)

# Calculate Alpha diversity(Richness)
richness_bac <- estimate_richness(physeq.r.bac, measures=c("Observed")) #the number of ASVs
colnames(richness_bac) <- "richness_bac"
richness_fungi <- estimate_richness(physeq.r.fungi, measures=c("Observed")) #the number of ASVs
colnames(richness_fungi) <- "richness_fungi"
data_base_richness <- merge(data_base,richness_bac, by = "row.names")
rownames(data_base_richness) <- data_base_richness$Row.names
data_base_richness <- merge(data_base_richness,richness_fungi, by = "row.names")
rownames(data_base_richness) <- data_base_richness$Row.names
data_base_richness <- data_base_richness[,-c(1:2)]
#write.csv(data_base_richness, here("dataanalysis/00data","data_base_richness.csv"))
```

```{r Microbial abundance vs soil}
# Run random forest
set.seed(123)

## Random forest for Bacterial abundance
RF_BacAbund <- data_base[,c("Bacterial.16S","Total.N","CN","pH.H2O.","WRB","Dominant_tree_type","MAP", "MAT")] # this dataset includes only the variables of interest
rfp_BacAbund <- rfPermute(Bacterial.16S~., data = RF_BacAbund, importance = TRUE, ntree = 500, nrep = 1000, num.cores = 1)
importance_BacAbund <- data.frame(Variable = rownames(importance(rfp_BacAbund, scale = TRUE)),
                                  importance(rfp_BacAbund, scale = TRUE), 
                                  heck.names = FALSE)

## Add a new column for asterisk labels based on the p-value condition
importance_BacAbund$label <- ifelse(importance_BacAbund$X.IncMSE.pval < 0.05, "*", "")
print(importance_BacAbund)

## Extract the last value of rfp_pH$rf$rsq, which corresponds to the overall % Var explained
var_explained_BacAbund <- tail(rfp_BacAbund$rf$rsq, 1) * 100
print(var_explained_BacAbund)

## Plot
RF_BacAbund_plot <-
  ggplot(importance_BacAbund, aes(x = reorder(Variable, X.IncMSE), y = X.IncMSE)) +
  geom_bar(stat = "identity", fill = "#56B4E9", alpha = 0.6) +
  geom_text(aes(label = label, y = X.IncMSE + 2), hjust = 1) + # Add stat
  coord_flip() + # Flip coordinates for horizontal bars
  labs(x = "", y = "Increase in MSE (%)", 
       subtitle = paste("Bacterial abundance\nVariation explained:", round(var_explained_BacAbund, 2), "%")) + # add var explained
  theme_classic() +
  theme(axis.title =  element_text(size = 12),
        axis.text = element_text(size = 12),
        plot.subtitle = element_text(size = 12)) + 
  scale_x_discrete(labels = function(x) sapply(x, function(el) {
    if (el == "WRB") return("Soil type")
    else if (el == "Dominant_tree_type") return("Vegetation")
    else if (el == "pH.H2O.") return("Soil habitat pH") # Change for soil pH
    else if (el == "Total.N") return("Soil N content") # Change for Soil N
    else if (el == "CN") return("Soil C-to-N ratio") # Change for Soil N
    else return(el)
  }))
print(RF_BacAbund_plot)

ggsave(paste0(Figure_path,"Figure2A_RF_BacAbund_plot.pdf"),RF_BacAbund_plot,width = 4,height = 4)

## Random forest for Fungal abundance
RF_FungAbund <- data_base[,c("Fungal.18S","Total.N","CN","pH.H2O.","WRB","Dominant_tree_type","MAP", "MAT")] # this dataset includes only the variables of interest
rfp_FungAbund <- rfPermute(Fungal.18S~., data = RF_FungAbund, importance = TRUE, ntree = 500, nrep = 1000, num.cores = 1)
importance_FungAbund <- data.frame(Variable = rownames(importance(rfp_FungAbund, scale = TRUE)),
                                   importance(rfp_FungAbund, scale = TRUE), 
                                   heck.names = FALSE)

## Add a new column for asterisk labels based on the p-value condition
importance_FungAbund$label <- ifelse(importance_FungAbund$X.IncMSE.pval < 0.05, "*", "")
print(importance_FungAbund)

## Extract the last value of rfp$rf$rsq, which corresponds to the overall % Var explained
var_explained_FungAbund <- tail(rfp_FungAbund$rf$rsq, 1) * 100
print(var_explained_FungAbund)

## Plot
RF_FungAbund_plot <-
  ggplot(importance_FungAbund, aes(x = reorder(Variable, X.IncMSE), y = X.IncMSE)) +
  geom_bar(stat = "identity", fill = "#CC79A7", alpha = 0.6) +
  geom_text(aes(label = label, y = X.IncMSE + 2), hjust = 1) + # Add stat
  coord_flip() + # Flip coordinates for horizontal bars
  labs(x = "", y = "Increase in MSE (%)", 
       subtitle = paste("Fungal abundance\nVariation explained:", round(var_explained_FungAbund, 2), "%")) + # add var explained
  theme_classic() +
  theme(axis.title =  element_text(size = 12),
        axis.text = element_text(size = 12),
        plot.subtitle = element_text(size = 12)) + 
  scale_x_discrete(labels = function(x) sapply(x, function(el) {
    if (el == "WRB") return("Soil type")
    else if (el == "Dominant_tree_type") return("Vegetation")
    else if (el == "pH.H2O.") return("Soil habitat pH") # Change for soil pH
    else if (el == "Total.N") return("Soil N content") # Change for Soil N
    else if (el == "CN") return("Soil C-to-N ratio") # Change for Soil N
    else return(el)
  }))
print(RF_FungAbund_plot)

ggsave(paste0(Figure_path,"Figure2B_RF_FungalAbund_plot.pdf"),RF_FungAbund_plot,width = 4,height = 4)

# Scatter plot of Abunadnce vs Total.N
data_long <- data_base %>%
  select(Total.N, Bacterial.16S, Fungal.18S) %>%
  pivot_longer(cols = -Total.N, names_to = "Taxa", values_to = "Abundance")

## Calculate statistics for Bacterial.16S
stats_bacterial <- cor.test(data_base$Total.N, data_base$Bacterial.16S, method = "pearson")

## Calculate statistics for Fungal.18S
stats_fungal <- cor.test(data_base$Total.N, data_base$Fungal.18S, method = "pearson")

## Adjust these y values based on your plot's data range and appearance
bacteria_annotation_y <- max(data_long$Abundance) * 10 # Adjusted for visual fit
fungi_annotation_y <- max(data_long$Abundance) * 4 # Slightly lower

## Plot
SoilNvsAbund_plot <- 
  ggplot(data_long, aes(x = Total.N, y = Abundance, color = Taxa)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) + # Add linear model fit lines
  theme_classic() +
  theme(axis.title = element_text(size = 12), 
        axis.text = element_text(size = 10),
        legend.position = "none") +
  labs(x = "Soil N content (%)", y = "Abundance", title = "") +
  scale_color_manual(values = c("Bacterial.16S" = "#56B4E9", "Fungal.18S" = "#CC79A7")) + # Custom colors
  annotation_logticks(sides = "b") + # Add log ticks on the bottom (X-axis)
  scale_x_log10(limits = c(0.1, 1.5), breaks = c(0.1, 0.5, 1, 1.5)) + # Log scale for X-axis
  annotation_logticks(sides = "l") + # Add log ticks on the left (Y-axis)
  scale_y_log10(limits = c(5e+7, 1e+13), breaks = 10^(8:12)) + # Log scale for Y-axis
  annotate("text", x = 0.1, y = bacteria_annotation_y, 
           label = sprintf("Bacteria: R = %.2f, P < 0.001", stats_bacterial$estimate),
           hjust = 0, size = 3.5, color = "#56B4E9") +
  annotate("text", x = 0.1, y = fungi_annotation_y, 
           label = sprintf("Fungi:     R = %.2f, P < 0.001", stats_fungal$estimate),
           hjust = 0, size = 3.5, color = "#CC79A7")
print(SoilNvsAbund_plot)

ggsave(paste0(Figure_path,"Figure2C_Scatter_BacFungalAbund_plot.pdf"),SoilNvsAbund_plot,width = 4,height = 4)
```


```{r Microbial richness vs soil}
# Run random forest
set.seed(123)

# Random forest for Bacterial richness
RF_BacRichness <- data_base_richness[,c("richness_bac","Total.N","CN","pH.H2O.","WRB","Dominant_tree_type","MAP", "MAT")] # this dataset includes only the variables of interest
rfp_BacRichness <- rfPermute(richness_bac~., data = RF_BacRichness, importance = TRUE, ntree = 500, nrep = 1000, num.cores = 1)
importance_BacRichness <- data.frame(Variable = rownames(importance(rfp_BacRichness, scale = TRUE)),
                                     importance(rfp_BacRichness, scale = TRUE), 
                                     heck.names = FALSE)

## Add a new column for asterisk labels based on the p-value condition
importance_BacRichness$label <- ifelse(importance_BacRichness$X.IncMSE.pval < 0.05, "*", "")
print(importance_BacRichness)

## Extract the last value of rfp$rf$rsq, which corresponds to the overall % Var explained
var_explained_BacRichness <- tail(rfp_BacRichness$rf$rsq, 1) * 100
print(var_explained_BacRichness)

## Plot
RF_BacRichness_plot <-
  ggplot(importance_BacRichness, aes(x = reorder(Variable, X.IncMSE), y = X.IncMSE)) +
  geom_bar(stat = "identity", fill = "#56B4E9", alpha = 0.6) +
  geom_text(aes(label = label, y = X.IncMSE + 2), hjust = 1) + # Add stat
  coord_flip() + # Flip coordinates for horizontal bars
  labs(x = "", y = "Increase in MSE (%)", 
       subtitle = paste("Bacterial taxomomic richness\nVariation explained:", round(var_explained_BacRichness, 2), "%")) + # add var explained
  theme_classic() +
  theme(axis.title =  element_text(size = 12),
        axis.text = element_text(size = 12),
        plot.subtitle = element_text(size = 12)) + 
  scale_x_discrete(labels = function(x) sapply(x, function(el) {
    if (el == "WRB") return("Soil type")
    else if (el == "Dominant_tree_type") return("Vegetation")
    else if (el == "pH.H2O.") return("Soil habitat pH") # Change for soil pH
    else if (el == "Total.N") return("Soil N content") # Change for Soil N
    else if (el == "CN") return("Soil C-to-N ratio") # Change for Soil N
    else return(el)
  }))
print(RF_BacRichness_plot)

ggsave(paste0(Figure_path,"Figure2G_RF_BacRichness_plot.pdf"),RF_BacRichness_plot,width = 4,height = 4)

# Random forest for Fungal richness
RF_FungRichness <- data_base_richness[,c("richness_fungi","Total.N","CN","pH.H2O.","WRB","Dominant_tree_type","MAP", "MAT")] # this dataset includes only the variables of interest
rfp_FungRichness <- rfPermute(richness_fungi~., data = RF_FungRichness, importance = TRUE, ntree = 500, nrep = 1000, num.cores = 1)
importance_FungRichness <- data.frame(Variable = rownames(importance(rfp_FungRichness, scale = TRUE)),
                                      importance(rfp_FungRichness, scale = TRUE), 
                                      heck.names = FALSE)
## Add a new column for asterisk labels based on the p-value condition
importance_FungRichness$label <- ifelse(importance_FungRichness$X.IncMSE.pval < 0.05, "*", "")
print(importance_FungRichness)

## Extract the last value of rfp$rf$rsq, which corresponds to the overall % Var explained
var_explained_FungRichness <- tail(rfp_FungRichness$rf$rsq, 1) * 100
print(var_explained_FungRichness)

## Plot
RF_FungRichness_plot <-
  ggplot(importance_FungRichness, aes(x = reorder(Variable, X.IncMSE), y = X.IncMSE)) +
  geom_bar(stat = "identity", fill = "#CC79A7", alpha = 0.6) +
  geom_text(aes(label = label, y = X.IncMSE + 2), hjust = 1) + # Add stat
  coord_flip() + # Flip coordinates for horizontal bars
  labs(x = "", y = "Increase in MSE (%)", 
       subtitle = paste("Fungal taxomomic richness\nVariation explained:", round(var_explained_FungRichness, 2), "%")) + # add var explained
  theme_classic() +
  theme(axis.title =  element_text(size = 12),
        axis.text = element_text(size = 12),
        plot.subtitle = element_text(size = 12)) + 
  scale_x_discrete(labels = function(x) sapply(x, function(el) {
    if (el == "WRB") return("Soil type")
    else if (el == "Dominant_tree_type") return("Vegetation")
    else if (el == "pH.H2O.") return("Soil habitat pH") # Change for soil pH
    else if (el == "Total.N") return("Soil N content") # Change for Soil N
    else if (el == "CN") return("Soil C-to-N ratio") # Change for Soil N
    else return(el)
  }))
print(RF_FungRichness_plot)

ggsave(paste0(Figure_path,"Figure2H_RF_FungalRichness_plot.pdf"),RF_FungRichness_plot,width = 4,height = 4)

# Scatter plot of Richness vs soil pH
data_base_richness_long <- data_base_richness %>%
  select(pH.H2O., richness_bac, richness_fungi) %>%
  pivot_longer(cols = -pH.H2O., names_to = "Taxa", values_to = "Richness")

## Calculate statistics for Bacterial richness
stats_bacterial <- cor.test(data_base_richness$pH.H2O., data_base_richness$richness_bac, method = "pearson")

## Calculate statistics for Fungal richness
stats_fungal <- cor.test(data_base_richness$pH.H2O., data_base_richness$richness_fungi, method = "pearson")

## Adjust these y values based on your plot's data range and appearance
bacteria_annotation_y <- max(data_base_richness_long$Richness) * 1.2 # Adjusted for visual fit
fungi_annotation_y <- max(data_base_richness_long$Richness) * 1.1 # Slightly lower

## Plot
pHvsRichnessVs_plot <- 
  ggplot(data_base_richness_long, aes(x = pH.H2O., y = Richness, color = Taxa)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) + # Add linear model fit lines
  theme_classic() +
  theme(axis.title = element_text(size = 12), 
        axis.text = element_text(size = 10),
        legend.position = "none") +
  labs(x = "Soil habitat pH", y = "Richness", title = "") +
  scale_y_continuous(limits = c(100, 9000), breaks = seq(0, 9000, by = 2000)) + # Adjust the y-axis scale
  scale_color_manual(values = c("richness_bac" = "#56B4E9", "richness_fungi" = "#CC79A7")) + # Custom colors
  annotate("text", x = 3.5, y = bacteria_annotation_y, 
           label = sprintf("Bacteria: R = %.2f, P < 0.001", stats_bacterial$estimate),
           hjust = 0, size = 3.5, color = "#56B4E9") +
  annotate("text", x = 3.5, y = fungi_annotation_y, 
           label = sprintf("Fungi:     R = %.2f, P < 0.001", stats_fungal$estimate),
           hjust = 0, size = 3.5, color = "#CC79A7")
print(pHvsRichnessVs_plot)

ggsave(paste0(Figure_path,"Figure2I_Scatter_BacFungalRichness_plot.pdf"),pHvsRichnessVs_plot,width = 4,height = 4)

```

```{r Microbial_composition_vs_soil}
# Use gdm_bac
head(gdm_bac)
head(gdm_fung)

# Reshape the data frame to long format
gdm_bac_long <- pivot_longer(gdm_bac, cols = starts_with("x."), names_to = "Variable", names_prefix = "x.", values_to = "X") %>%
  pivot_longer(cols = starts_with("y."), names_to = "Variable_y", names_prefix = "y.", values_to = "Y") %>%
  filter(substring(Variable, 1, nchar(Variable)) == substring(Variable_y, 1, nchar(Variable_y))) %>%
  mutate(Variable = substring(Variable, 1, nchar(Variable)))

gdm_fung_long <- pivot_longer(gdm_fung, cols = starts_with("x."), names_to = "Variable", names_prefix = "x.", values_to = "X") %>%
  pivot_longer(cols = starts_with("y."), names_to = "Variable_y", names_prefix = "y.", values_to = "Y") %>%
  filter(substring(Variable, 1, nchar(Variable)) == substring(Variable_y, 1, nchar(Variable_y))) %>%
  mutate(Variable = substring(Variable, 1, nchar(Variable)))

# Plot
# Set the order of the factors
gdm_bac_long$Variable <- factor(gdm_bac_long$Variable, 
                                levels = c("pH","CN","MAT","TC","dist","tree","soil","MAP"))
# Custom colors for each variable
custom_colors <- c(MAT = "#1f77b4", MAP = "#ff7f0e", pH = "#2ca02c", TC = "#d62728", 
                   CN = "#9467bd", tree = "#8c564b", soil = "#e377c2", dist = "#7f7f7f")
# Plot with ordered legend and 'Paired' color palette
GDM_bac_plot <- 
  ggplot(gdm_bac_long, aes(x = X, y = Y, color = Variable)) +
  geom_line() +
  scale_color_manual(values = custom_colors,
                     labels = c("Soil habitat pH", "Soil C-to-N ratio",
                                "MAT", "Soil C content", 
                                "Geographic dist.", "Vegetation", 
                                "Soil type", "MAP")) +
  theme_classic() +
  theme(axis.title = element_text(size = 12),
        axis.text = element_text(size = 8),
        plot.subtitle = element_text(size = 12),
        legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  labs(x = "Predictor dissimilarity", y = "Compositional dissimilarity",
       subtitle = "Bacterial taxomomic composition\nVariation explained: 61.72%",
       color = "Variable")
print(GDM_bac_plot)
ggsave(paste0(Figure_path,"Figure2D_GDM_Bac_plot.pdf"),GDM_bac_plot,width = 4,height = 4)

# Set the order of the factors
gdm_fung_long$Variable <- factor(gdm_fung_long$Variable, 
                                levels = c("pH","dist","MAT","tree","CN","MAP","TC","soil"))
# Custom colors for each variable
custom_colors <- c(MAT = "#1f77b4", MAP = "#ff7f0e", pH = "#2ca02c", TC = "#d62728", 
                   CN = "#9467bd", tree = "#8c564b", soil = "#e377c2", dist = "#7f7f7f")
# Plot with ordered legend and 'Paired' color palette
GDM_fung_plot <- 
  ggplot(gdm_fung_long, aes(x = X, y = Y, color = Variable)) +
  geom_line() +
  scale_color_manual(values = custom_colors,
                     labels = c("Soil habitat pH","Geographic dist.","MAT","Vegetation",
                                "Soil C-to-N ratio","MAP","Soil C content","Soil type")) +
  theme_classic() +
  theme(axis.title = element_text(size = 12),
        axis.text = element_text(size = 8),
        plot.subtitle = element_text(size = 12),
        legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  labs(x = "Predictor dissimilarity", y = "Compositional dissimilarity",
       subtitle = "Fungal taxomomic composition\nVariation explained: 38.01%",
       color = "Variable")
print(GDM_fung_plot)
ggsave(paste0(Figure_path,"Figure2E_GDM_Fungi_plot.pdf"),GDM_fung_plot,width = 4,height = 4)


# Scatter plot of Composition vs soil pH
## Select and format data
data_base_190 <- read.csv(data_path_190, header = TRUE, sep = "\t", row.names = 1)
data_asv_nmds <- data_base_190 %>%
  select(pH.H2O., NMDS1_bac, NMDS1_fungi) %>%
  pivot_longer(cols = -pH.H2O., names_to = "Taxa", values_to = "Composition")

## Calculate statistics for Bacterial.16S
stats_bacterial_nmds <- cor.test(data_base_190$pH.H2O., data_base_190$NMDS1_bac, method = "pearson")

## Calculate statistics for Fungal.18S
stats_fungal_nmds <- cor.test(data_base_190$pH.H2O., data_base_190$NMDS1_fungi, method = "pearson")

## Adjust these y values based on your plot's data range and appearance
bacteria_annotation_y <- max(data_asv_nmds$Composition) * 1.3 # Adjusted for visual fit
fungi_annotation_y <- max(data_asv_nmds$Composition) * 1.1 # Slightly lower

## Plot
pHvsComposition_plot <- 
  ggplot(data_asv_nmds, aes(x = pH.H2O., y = Composition, color = Taxa)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) + # Add linear model fit lines
  theme_classic() +
  theme(axis.title = element_text(size = 12), 
        axis.text = element_text(size = 10),
        legend.position = "none") +
  labs(x = "Soil habitat pH", y = "Composition (MDS1)", title = "") +
  #scale_y_continuous(limits = c(100, 14000), breaks = seq(0, 12000, by = 4000)) + # Adjust the y-axis scale
  scale_color_manual(values = c("NMDS1_bac" = "#56B4E9", "NMDS1_fungi" = "#CC79A7")) + # Custom colors
  annotate("text", x = 3.5, y = bacteria_annotation_y, 
           label = sprintf("Bacteria: R = %.2f, P < 0.001", stats_bacterial_nmds$estimate),
           hjust = 0, size = 3.5, color = "#56B4E9") +
  annotate("text", x = 3.5, y = fungi_annotation_y, 
           label = sprintf("Fungi:     R = %.2f, P < 0.001", stats_fungal_nmds$estimate),
           hjust = 0, size = 3.5, color = "#CC79A7")
print(pHvsComposition_plot)


# Save
ggsave(paste0(Figure_path, "Figure2F_pHvsComposition_plot.pdf"), 
       pHvsComposition_plot, width = 4, height = 4, dpi = 300)

```


```{r save sub-figures of figure2}
#ggsave(paste0(figure2_path, "/RF_BacAbund_plot.png"), RF_BacAbund_plot, width = 4, height = 3, dpi = 300)
#ggsave(paste0(figure2_path, "/RF_FungAbund_plot.png"), RF_FungAbund_plot, width = 4, height = 3, dpi = 300)
#ggsave(paste0(figure2_path, "/SoilNvsAbund_plot.png"), SoilNvsAbund_plot, width = 3, height = 3, dpi = 300)
#ggsave(paste0(figure2_path, "/RF_BacRichness_plot.png"), RF_BacRichness_plot, width = 4, height = 3, dpi = 300)
#ggsave(paste0(figure2_path, "/RF_FungRichness_plot.png"), RF_FungRichness_plot, width = 4, height = 3, dpi = 300)
#ggsave(paste0(figure2_path, "/pHvsRichnessVs_plot.png"), pHvsRichnessVs_plot, width = 3, height = 3, dpi = 300)

```

>**Figure3: Cascading relationship from landscape factors, soil habitat properties, and microbiome to N cycling**

```{r calculate the number of KO genes and n cycling genes}
# Rarefy OTU table
## N cycling genes
set.seed(99)
physeq.r.N <- rarefy_even_depth(physeq_N, sample.size =min(sample_sums(physeq_N)),rngseed = FALSE, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
physeq.r.N
#write.table(otu_table(physeq.r.N), file = here("dataanalysis/00data/rawdata","N-cycling-38.txt"),sep = "\t",row.names = T,col.names = NA,quote = F)
physeq.r.KO <- rarefy_even_depth(physeq_KO, sample.size =min(sample_sums(physeq_KO)),rngseed = FALSE, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
physeq.r.KO
#write.table(otu_table(physeq.r.KO), file = here("dataanalysis/00data/rawdata","KO-38.txt"),sep = "\t",row.names = T,col.names = NA,quote = F)

# Calculate Alpha diversity(Richness)
richness_Ngene <- estimate_richness(physeq.r.N, measures=c("Observed")) #the number of ASVs
colnames(richness_Ngene) <- "richness_N"
richness_KOgene <- estimate_richness(physeq.r.KO, measures=c("Observed")) #the number of ASVs
colnames(richness_KOgene) <- "richness_KO"

data_N_richness <- merge(data_base_38,richness_Ngene, by = "row.names")
rownames(data_N_richness) <- data_N_richness$Row.names
data_N_richness <- merge(data_N_richness,richness_KOgene, by = "row.names")
rownames(data_N_richness) <- data_N_richness$Row.names
data_N_richness <- data_N_richness[,-c(1:2)]

```


```{r Figure3B_SEM_alpha}
#data for SEM_alpha(no data: -TMC, -TMN)
data_N_richness[is.na(data_N_richness)]<-0
colnames(data_N_richness)
data_N_richness <- data_N_richness %>%
  mutate(
    GMIN_GNIT_GMIN_reverse = -1 * GMIN_GNIT_GMIN
  )
#write.csv(data_N_richness, here("dataanalysis/00data","data_N_richness.csv"))

data_N_richness_scale <- data_N_richness
data_N_richness_scale[,c("Dominant_tree_type_1","soil_type","pH.H2O.",'Total.C', 'Total.N',"richness_bac", "richness_fungi","richness_N","MAT","MAP","GNIT_GMIN","GMIN_GNIT_GMIN","GMIN","GNIT","GNH4C","GNO3C","GMIN_GNIT_GMIN_reverse")]<- scale(data_N_richness[,c("Dominant_tree_type_1","soil_type","pH.H2O.",'Total.C', 'Total.N',"richness_bac", "richness_fungi","richness_N","MAT","MAP","GNIT_GMIN","GMIN_GNIT_GMIN","GMIN","GNIT","GNH4C","GNO3C","GMIN_GNIT_GMIN_reverse")])

#Specifies the latent variable and stores the relationship between the variable and the latent variable as a list in R
dat_blocks_alpha <- list(
  Climate = c("MAT","MAP"),
  Vegetation="Dominant_tree_type_1",
  Soiltype="soil_type",
  pH="pH.H2O.",
  SOM= c('Total.C', 'Total.N'), 
  richness = c('richness_fungi','richness_bac'),
  N_cycling_richness="richness_N",
  Rate=c("GMIN","GNIT","GNH4C","GNO3C"),
  Openess=c('GNIT_GMIN','GMIN_GNIT_GMIN_reverse'))
#The association between latent variables is described by 0-1 matrix, where 0 means that there is no association between variables and 1 means that there is association
Climate <- c(0,0, 0, 0, 0, 0,0, 0,0)
Vegetation <- c(0,0, 0, 0, 0, 0,0, 0,0)
Soiltype <- c(0,0, 0, 0, 0, 0,0, 0,0)
pH<- c(1,1, 1, 0, 0, 0,0, 0,0)
SOM<- c(1,1, 1, 0, 0, 0,0, 0,0)
richness <- c(0,0, 0, 1, 1, 0,0, 0,0)
N_cycling_richness<- c(0,0, 0, 0, 0, 1,0, 0,0)
Rate<- c(0,0, 0, 0, 0, 1,0, 0,0)
Openess<- c(0,0, 0, 0, 0, 1,0, 0,0)
dat_path_alpha <- rbind(Climate,Vegetation, Soiltype, pH, SOM, richness, N_cycling_richness,Rate,Openess)
colnames(dat_path_alpha) <- rownames(dat_path_alpha)
#innerplot can see if the association is correct
innerplot(dat_path_alpha)
#Specify the direction of variables, optionally A (for the column is the cause of the row) or B (for the row is the cause of the column)
dat_modes_alpha <- c("B","B","B","A", "A","A","A","A","A")
dat_pls_alpha <- plspm(data_N_richness_scale, dat_path_alpha, dat_blocks_alpha, modes = dat_modes_alpha,boot.val = TRUE,br=1000)
summary(dat_pls_alpha)

```

```{r Figure3A_sem_beta}
data_base_190[is.na(data_base_190)]<-0
data_base_190$GMIN_GNIT_GMIN <- data_base_190$GMIN_GNIT_GMIN*-1

data_base_190[,c(2:16,18:41,44)]<- scale(data_base_190[,c(2:16,18:41,44)])

data_base_190 <- na.omit(data_base_190)
library(dplyr)
data_base_190 <- data_base_190 %>% mutate_if(is.character, as.factor)
str(data_base_190)
## SEM
#Specify latent variables.
colnames(data_base_190)
dat_blocks <- list(
  treetype="Dominant_tree_type_1",
  soiltype='soil_type',
  climate=c('MAT','MAP'),
  pH="pH.H2O.",
  SOM=c("Total.C",'Total.N'),
  NMDS1_bac='NMDS1_bac',
  NMDS1_fungi='NMDS1_fungi',
  Bacterialabundance= 'Bacterial.16S',
  Openness=c('GNIT_GMIN','GMIN_GNIT_GMIN'),
  NCR=c("GMIN","GNIT","GNH4C","GNO3C"))

dat_blocks
#The association between latent variables is described through a 0-1 matrix, where 0 indicates no association between the variables and 1 indicates an association.
treetype <- c(0, 0, 0, 0, 0,0,0, 0, 0, 0)
soiltype<- c(0, 0, 0, 0, 0,0,0, 0, 0, 0)
climate<- c(0, 0, 0, 0, 0,0,0, 0, 0, 0)
pH<-c(1, 1, 1,0, 0,0,0, 0, 0, 0)
SOM<- c(1,1, 1, 0, 0,0,0, 0, 0, 0)
NMDS1_bac<- c(0, 0,0,1,1,0, 0, 0, 0, 0)
NMDS1_fungi<- c(0, 0,0,1,1,0, 0, 0,0,  0)
Bacterialabundance<- c(0, 0,0,1,1,0,0, 0,  0, 0)
Openness<- c(0, 0, 0, 0, 0,1,1,1,0, 0)
NCR<-c(0, 0, 0, 0, 0,1,1,1,0, 0)

dat_path <- rbind(treetype,soiltype,climate,pH,SOM,NMDS1_bac,NMDS1_fungi,Bacterialabundance,Openness,NCR)

colnames(dat_path) <- rownames(dat_path)
dat_path
#innerplot can be used to check if the relationship diagram is correct
innerplot(dat_path)
dat_modes <- c("B","B","A","A","A","A", "A","A","A", "A")
dat_modes
dat_pls <- plspm(data_base_190, dat_path, dat_blocks, modes = dat_modes,boot.val = TRUE,br=1000)
summary(dat_pls)

```


```{r Figure3C_bacterial abundance vs. N cycling rate}

head(data_base) 

# Replace negative values with zero
data_base <- data_base %>%
  mutate(GMIN = ifelse(GMIN < 0, 0, GMIN),
         GNIT = ifelse(GNIT < 0, 0, GNIT),
         GNH4C = ifelse(GNH4C < 0, 0, GNH4C),
         GNO3C = ifelse(GNO3C < 0, 0, GNO3C))

## Standardize the data
data_base <- data_base %>%
  mutate(across(c(GMIN, GNIT, GNH4C, GNO3C), ~scale(.)[, 1], .names = "std_{.col}"))

## Merge as "N-cycling rate" by calculating the mean of standardized values for each row
data_base <- data_base %>%
  rowwise() %>%
  mutate(N_cycling_rate = mean(c_across(starts_with("std_")), na.rm = TRUE)) %>%
  ungroup()

## Check the resulting data
glimpse(data_base)

## Calculate statistics for Bacterial.16S
stats_bacterial_rate <- cor.test(data_base$Bacterial.16S, data_base$N_cycling_rate, method = "pearson")
stats_fungal_rate <- cor.test(data_base$Fungal.18S, data_base$N_cycling_rate, method = "pearson")

## Plot
RatevsAbundance_plot <- 
  ggplot(data_base, aes(x = Bacterial.16S, y = N_cycling_rate)) +  # Set common aes
  geom_point(aes(fill = Total.N), shape = 21, colour = "#56B4E9", size = 2) +  # Set fill here
  geom_smooth(method = "lm", se = FALSE, aes(group = 1), color = "#56B4E9") +  # Add 'group = 1' if needed
  theme_classic() +
  theme(axis.title = element_text(size = 12), 
        axis.text = element_text(size = 10),
        legend.position = "right") +
  labs(x = "Bacterial abundance (/g-soil)", y = "N-cycling rate (standardized)", fill = " Soil N content\n(%)", title = "") +
  scale_fill_gradient(low = "white", high = "#56B4E9") +  # Apply gradient fill to points 
  scale_y_continuous(limits = c(-1, 5), breaks = seq(-1, 4, by = 1)) + 
  annotate("text", x = 0, y = 4.5, 
           label = sprintf("Bacteria: R = %.2f, P < 0.001", stats_bacterial_rate$estimate),
           hjust = 0, size = 3.5, color = "#56B4E9") 
print(RatevsAbundance_plot)

# Save
ggsave(paste0(Figure_path, "Figure3C_BacAbundNRate.pdf"), 
       RatevsAbundance_plot, width = 4, height = 3, dpi = 300)
```


```{r Figure3D_Microbial composition vs. N allocation}
data_base_190 <- read.csv(data_path_190, header = TRUE, sep = "\t", row.names = 1)
# Replace negative values with zero
data_base_190 <- data_base_190 %>%
  mutate(GMIN = ifelse(GMIN < 0, 0, GMIN),
         GNIT = ifelse(GNIT < 0, 0, GNIT),
         GNH4C = ifelse(GNH4C < 0, 0, GNH4C),
         GNO3C = ifelse(GNO3C < 0, 0, GNO3C))

# Calculate new ratios
## equation 1
## nitrification (GNIT) divided by ammonification (GMIN)
## higher -> open
## equation 2
## (ammonification (GMIN) - nitrification (GNIT)) divided by ammonification (GMIN)
## higher -> closed

data_base_190 <- data_base_190 %>%
  mutate(
    GNIT_div_GMIN = GNIT / GMIN,
    GMIN_minus_GNIT_div_GMIN = ((GMIN - GNIT) / GMIN)
  )

data_base_190 <- data_base_190 %>%
  mutate(
    GMIN_minus_GNIT_div_GMIN = ifelse(GMIN_minus_GNIT_div_GMIN < 0, 0, GMIN_minus_GNIT_div_GMIN),
    GMIN_minus_GNIT_div_GMIN_rev = 1 - GMIN_minus_GNIT_div_GMIN
  )

data_asv_allocation <- data_base_190 %>%
  mutate(
    std_GNIT_div_GMIN = scale(GNIT_div_GMIN, center = TRUE, scale = TRUE)[,1],
    std_GMIN_minus_GNIT_div_GMIN = scale(GMIN_minus_GNIT_div_GMIN, center = TRUE, scale = TRUE)[,1],
    std_GMIN_minus_GNIT_div_GMIN_rev = scale(GMIN_minus_GNIT_div_GMIN_rev, center = TRUE, scale = TRUE)[,1]
  )

## Merge as "N allocation" by calculating the mean of standardized values for each row
data_asv_allocation <- data_asv_allocation %>%
  rowwise() %>%
  mutate(
    N_allocation = mean(c(std_GNIT_div_GMIN, std_GMIN_minus_GNIT_div_GMIN_rev), na.rm = TRUE)
  ) %>%
  ungroup()

## Check the resulting data
glimpse(data_asv_allocation)

## Calculate statistics with standarized value
stats_bacterial_allocation_std <- cor.test(data_asv_allocation$NMDS1_bac, data_asv_allocation$N_allocation, method = "pearson")
stats_fungal_allocation_std <- cor.test(data_asv_allocation$NMDS1_fungi, data_asv_allocation$N_allocation, method = "pearson")

## Plot with standarized value
Allocation_stdvsBacComposition_plot <- 
  ggplot(data_asv_allocation, aes(x = NMDS1_bac, y = N_allocation)) +  # Set common aes
  geom_point(aes(fill = pH.H2O.), shape = 21, colour = "#56B4E9", size = 2) +  # Set fill here
  geom_smooth(method = "lm", se = FALSE, aes(group = 1), color = "#56B4E9") +  # Add 'group = 1' if needed
  theme_classic() +
  theme(axis.title = element_text(size = 12), 
        axis.text = element_text(size = 10),
        legend.position = "right") +
  labs(x = "MDS1 (Bacteria)", y = "N allocation (standarized)", fill = " Soil pH", title = "") +
  scale_fill_gradient(low = "white", high = "#56B4E9") +  # Apply gradient fill to points 
  scale_y_continuous(limits = c(-1, 3), breaks = seq(-1, 3, by = 1)) + 
  annotate("text", x = -2, y = 3, 
           label = sprintf("Bacteria: R = %.2f, P = %.3f", 
                           stats_bacterial_allocation_std$estimate, 
                           stats_bacterial_allocation_std$p.value),
           hjust = 0, size = 3.5, color = "#56B4E9") 
print(Allocation_stdvsBacComposition_plot)

# Save
ggsave(paste0(Figure_path, "Figure3D_BacCompNAlloc.pdf"), 
       Allocation_stdvsBacComposition_plot, width = 3.5, height = 3, dpi = 300)

```


```{r Figure3E_bacterial richness vs. N genes richness}
## Calculate statistics for Bacterial.16S
stats_bacterial_N <- cor.test(data_N_richness$richness_bac, data_N_richness$richness_N, method = "pearson")

## Plot
BacvsN_Richness <- 
  ggplot(data_N_richness, aes(x = richness_bac, y = richness_N)) +  # Set common aes
  geom_point(aes(fill = pH.H2O.), shape = 21, colour = "#56B4E9", size = 2) +  # Set fill here
  geom_smooth(method = "lm", se = FALSE, aes(group = 1), color = "#56B4E9") +  # Add 'group = 1' if needed
  theme_classic() +
  theme(axis.title = element_text(size = 12), 
        axis.text = element_text(size = 10),
        legend.position = "right") +
  labs(x = "Richness (Bacteria)", y = "N-cycling gene richness", fill = " Soil pH", title = "") +
  scale_fill_gradient(low = "white", high = "#56B4E9") +  # Apply gradient fill to points 
  #scale_y_continuous(limits = c(-1, 5), breaks = seq(-1, 4, by = 1)) + 
  annotate("text", x = 2900, y = 29, 
           label = sprintf("Bacteria: R = %.2f, P < 0.001", stats_bacterial_N$estimate),
           hjust = 0, size = 3.5, color = "#56B4E9") 
print(BacvsN_Richness)

# Save
ggsave(paste0(Figure_path, "Figure3E_BacNRichness.pdf"), 
       BacvsN_Richness, width = 3.5, height = 3, dpi = 300)
```

>**Figure4: Context-dependent effects of microbial community components on N allocation**

```{r}
data_base_190 <- read.csv(data_path_190, header = TRUE, sep = "\t", row.names = 1)

data_base_190[is.na(data_base_190)]<-0
data_base_190$GMIN_GNIT_GMIN <- data_base_190$GMIN_GNIT_GMIN*-1

data_base_190[,c(2:16,18:41,44)]<- scale(data_base_190[,c(2:16,18:41,44)])

data_base_190 <- na.omit(data_base_190)


#subset
#prepare data for all vegetations and soil types.
#data for SEM_Cambisol
data_sem_Andosol <- subset(data_base_190,WRB_3!='Cambisols'&WRB_3!='others')
#data for SEM_Andosol
data_sem_Cambisol <- subset(data_base_190,WRB_3!='Andosols'&WRB_3!='others')
#data for SEM_Broadleaf
data_sem_Broadleaf <- subset(data_base_190,Dominant_tree_type!='Conifer')
#data for SEM_Conifer
data_sem_Conifer <- subset(data_base_190,Dominant_tree_type!='Broadleaf')

# Figure 4A for broadleaved forest
colnames(data_base_190)
dat_blocks <- list(
  soiltype='soil_type',
  climate=c('MAT','MAP'),
  pH="pH.H2O.",
  SOM=c("Total.C",'Total.N'),
  NMDS1_bac='NMDS1_bac',
  NMDS1_fungi='NMDS1_fungi',
  Bacterialabundance= 'Bacterial.16S',
  Openness=c('GNIT_GMIN','GMIN_GNIT_GMIN'),
  NCR=c("GMIN","GNIT","GNH4C","GNO3C"))

dat_blocks
#The association between latent variables is described through a 0-1 matrix, where 0 indicates no association between the variables and 1 indicates an association.
soiltype <- c(0, 0, 0, 0, 0,0,0, 0, 0)
climate<- c(0, 0, 0, 0, 0,0,0, 0, 0)
pH<-c(1, 1, 0, 0,0,0, 0, 0, 0)
SOM<- c(1,0, 0, 0,0,0, 0, 0, 0)
NMDS1_bac<- c(0, 0,1,1,0,0, 0, 0, 0)
NMDS1_fungi<- c(0, 0,1, 1,0,0, 0,0,0)
Bacterialabundance<- c(0, 0,1, 1, 0,0,0,0, 0)
Openness<- c(0, 0, 0, 0,1,1,1, 0,0)
NCR<-c(0, 0, 0, 0,1,1,1,0,0)

dat_path <- rbind(soiltype,climate,pH,SOM,NMDS1_bac,NMDS1_fungi,Bacterialabundance,Openness,NCR)

colnames(dat_path) <- rownames(dat_path)
dat_path
innerplot(dat_path)
dat_modes <- c("B","A","A","A","A", "A","A","A","A")
dat_modes
dat_pls <- plspm(data_sem_Broadleaf, dat_path, dat_blocks, modes = dat_modes,boot.val = TRUE,br=1000)
summary(dat_pls)

# Figure 4B for Coniferous forest
colnames(data_base_190)
dat_blocks <- list(
  soiltype='soil_type',
  climate=c('MAT','MAP'),
  pH="pH.H2O.",
  SOM=c("Total.C",'Total.N'),
  NMDS1_bac='NMDS1_bac',
  NMDS1_fungi='NMDS1_fungi',
  Bacterialabundance= 'Bacterial.16S',
  Openness=c('GNIT_GMIN','GMIN_GNIT_GMIN'),
  NCR=c("GMIN","GNIT","GNH4C","GNO3C"))

dat_blocks
#The association between latent variables is described through a 0-1 matrix, where 0 indicates no association between the variables and 1 indicates an association.
soiltype <- c(0, 0, 0, 0, 0,0,0, 0, 0)
climate<- c(0, 0, 0, 0, 0,0,0, 0, 0)
pH<-c(1, 1, 0, 0,0,0, 0, 0, 0)
SOM<- c(1,1, 0, 0,0,0, 0, 0, 0)
NMDS1_bac<- c(0, 0,1,1,0,0, 0, 0, 0)
NMDS1_fungi<- c(0, 0,1, 1,0,0, 0,0,0)
Bacterialabundance<- c(0, 0,1, 1, 0,0,0,0, 0)
Openness<- c(0, 0, 0, 0,1,1,1, 0,0)
NCR<-c(0, 0, 0, 0,1,1,1,0,0)

dat_path <- rbind(soiltype,climate,pH,SOM,NMDS1_bac,NMDS1_fungi,Bacterialabundance,Openness,NCR)

colnames(dat_path) <- rownames(dat_path)
dat_path
innerplot(dat_path)
dat_modes <- c("B","A","A","A","A", "A","A","A","A")
dat_modes

dat_pls <- plspm(data_sem_Conifer, dat_path, dat_blocks, modes = dat_modes,boot.val = TRUE,br=1000)
summary(dat_pls)

# Figure 4c for Andosal forest
colnames(data_base_190)
dat_blocks <- list(
  treetype="Dominant_tree_type_1",
  climate=c('MAT','MAP'),
  pH="pH.H2O.",
  SOM=c("Total.C",'Total.N'),
  NMDS1_bac='NMDS1_bac',
  NMDS1_fungi='NMDS1_fungi',
  Bacterialabundance= 'Bacterial.16S',
  Openness=c('GNIT_GMIN','GMIN_GNIT_GMIN'),
  NCR=c("GMIN","GNIT","GNH4C","GNO3C"))

dat_blocks
#The association between latent variables is described through a 0-1 matrix, where 0 indicates no association between the variables and 1 indicates an association.
treetype <- c(0, 0, 0, 0, 0,0,0, 0, 0)
climate<- c(0, 0, 0, 0, 0,0,0, 0, 0)
pH<-c(1, 1, 0, 0,0,0, 0, 0, 0)
SOM<- c(1,1, 0, 0,0,0, 0, 0, 0)
NMDS1_bac<- c(0, 0,1,1,0,0, 0, 0, 0)
NMDS1_fungi<- c(0, 0,1, 1,0,0, 0,0,0)
Bacterialabundance<- c(0, 0,1, 1, 0,0,0,0, 0)
Openness<- c(0, 0, 0, 0,1,1,1, 0,0)
NCR<-c(0, 0, 0, 0,1,1,1,0,0)

dat_path <- rbind(treetype,climate,pH,SOM,NMDS1_bac,NMDS1_fungi,Bacterialabundance,Openness,NCR)

colnames(dat_path) <- rownames(dat_path)
dat_path
innerplot(dat_path)
dat_modes <- c("B","A","A","A","A", "A","A","A","A")
dat_modes
dat_pls <- plspm(data_sem_Andosol, dat_path, dat_blocks, modes = dat_modes,boot.val = TRUE,br=1000)
dat_pls
summary(dat_pls)

# Figure 4D for Cambisoal forest
colnames(data_base_190)
dat_blocks <- list(
  treetype="Dominant_tree_type_1",
  climate=c('MAT','MAP'),
  pH="pH.H2O.",
  SOM=c("Total.C",'Total.N'),
  NMDS1_bac='NMDS1_bac',
  NMDS1_fungi='NMDS1_fungi',
  Bacterialabundance= 'Bacterial.16S',
  Openness=c('GNIT_GMIN','GMIN_GNIT_GMIN'),
  NCR=c("GMIN","GNIT","GNH4C","GNO3C"))

dat_blocks
#The association between latent variables is described through a 0-1 matrix, where 0 indicates no association between the variables and 1 indicates an association.
treetype <- c(0, 0, 0, 0, 0,0,0, 0, 0)
climate<- c(0, 0, 0, 0, 0,0,0, 0, 0)
pH<-c(1, 1, 0, 0,0,0, 0, 0, 0)
SOM<- c(1,1, 0, 0,0,0, 0, 0, 0)
NMDS1_bac<- c(0, 0,1,1,0,0, 0, 0, 0)
NMDS1_fungi<- c(0, 0,1, 1,0,0, 0,0,0)
Bacterialabundance<- c(0, 0,1, 1, 0,0,0,0, 0)
Openness<- c(0, 0, 0, 0,1,1,1, 0,0)
NCR<-c(0, 0, 0, 0,1,1,1,0,0)

dat_path <- rbind(treetype,climate,pH,SOM,NMDS1_bac,NMDS1_fungi,Bacterialabundance,Openness,NCR)

colnames(dat_path) <- rownames(dat_path)
dat_path
innerplot(dat_path)
dat_modes <- c("B","A","A","A","A", "A","A","A","A")
dat_modes
dat_pls <- plspm(data_sem_Cambisol, dat_path, dat_blocks, modes = dat_modes,boot.val = TRUE,br=1000)
dat_pls
summary(dat_pls)
```

>**Figure5: Changes in the metabolic capabilities of the microbial community in response to changes in the soil pH. **

```{r}

# get base data
data_base <- read.csv(data_path)

# get module data
data_module_path <- "C:/Users/alip/Dropbox/Yaping/GRENE/dataanalysis/00data/module_table.txt"
data_module <- read.csv(data_module_path, header = TRUE, sep = "\t")

# get module ID list
data_module_ID_path <- "C:/Users/alip/Dropbox/Yaping/GRENE/dataanalysis/00data/KEGG_module_ID.txt"
data_module_ID <- read.csv(data_module_ID_path, header = TRUE, sep = "\t")

# Reshape data_module to long format
data_module_long <- pivot_longer(data_module, 
                                 cols = -Module, 
                                 names_to = "sample", 
                                 values_to = "module_abundance")
head(data_module_long)

# Merge with environment data 
data_module_merged <- merge(data_module_long, data_base, by = "sample") %>%
  filter(!is.na(module_abundance))
head(data_module_merged)


##########################################
# Spearman for each module
##########################################
# Run Spearman
module_spearman <- 
  data_module_merged %>%
  group_by(Module) %>%
  summarise(cor_value = cor.test(module_abundance, pH.H2O., method = "spearman", use = "complete.obs")$estimate,
            p_value = cor.test(module_abundance, pH.H2O., method = "spearman", use = "complete.obs")$p.value,
            mean_abundance = mean(module_abundance, na.rm = TRUE)) %>%
  filter(!is.na(cor_value) & cor_value != 0) %>%  # Remove rows where cor_value is NA or 0
  ungroup() 
head(module_spearman)

# Merge with module ID list
module_spearman <- merge(module_spearman, data_module_ID, by.x = "Module", by.y = "ID")
head(module_spearman)

# Organize and sort the data
module_spearman_table <- module_spearman %>%
  arrange(Type, Large_category, Small_category, desc(cor_value)) %>%
  select(Type, Large_category, Small_category, Module, Name, cor_value, p_value, mean_abundance, Origin)

# Save
#write.csv(module_spearman_table, paste0(save_plot_path, "/module_spearman_results.csv"), row.names = FALSE)

##########################################
# Filter
##########################################
# Filter the modules relating to N metabolisms and environmental infomation processing
module_spearman_filtered <- module_spearman_table %>%
  filter(
    (Type == "Pathway" & Large_category == "Energy metabolism") & Small_category == "Nitrogen metabolism" |
    (Type == "Complex" & Large_category == "Environmental information processing" & Small_category == "Mineral and organic ion transport system") |
    (Type == "Complex" & Large_category == "Environmental information processing" & Small_category == "Phosphate and amino acid transport system") |
    (Type == "FuncSet" & Large_category == "Environmental information processing" & Small_category == "Two-component regulatory system")
  )
head(module_spearman_filtered)

# Choose Pathway and Complex, and create the plot
module_pathway_complex <- 
  module_spearman_filtered %>%
  filter(Type == "Pathway" | Type == "Complex",
         p_value < 0.05) %>%
  arrange(desc(cor_value))

# Delete term after "," in the Name column
module_pathway_complex$Name <- gsub(",.*", "", module_pathway_complex$Name)
module_pathway_complex$Name <- gsub(" system", "", module_pathway_complex$Name)

# Set the desired order
desired_order <- c("Nitrogen metabolism", "Phosphate and amino acid transport system", "Mineral and organic ion transport system")
module_pathway_complex$Small_category <- factor(module_pathway_complex$Small_category, levels = desired_order)

# Order the data frame by Small_category (in the desired display order) and then by cor_value
module_pathway_complex <- module_pathway_complex %>%
  arrange(desc(Small_category), cor_value) %>%
  mutate(Name = factor(Name, levels = unique(Name))) # This relevels Name based on the arranged order

# Plot
module_pathway_complex_plot <-
  ggplot(module_pathway_complex, aes(x = Name, y = cor_value, fill = Small_category)) +
  geom_bar(stat = "identity") +
  coord_flip() + # Flips the axes, affecting how the factor levels are displayed
  labs(y = "Spearman correlation", x = "", title = "") +
  theme_classic() +
  theme(axis.title = element_text(size = 12),
        axis.text = element_text(size = 12),
        legend.text = element_text(size = 12),
        legend.position = "bottom", 
        legend.title = element_blank()) +
  scale_y_continuous(limits = c(-1, 1)) + # Adjust the y-axis scale
  scale_fill_brewer(palette = "Set1") + 
  guides(fill = guide_legend(nrow = 3, byrow = TRUE, title.position = 'left'))

# Choose FuncSet
module_funcset <- 
  module_spearman_filtered %>%
  filter(Type == "FuncSet",
         p_value < 0.05) %>%
  arrange(desc(cor_value))

# Delete "two-component regulatory system" in the Name column
module_funcset$Name <- gsub(" two-component regulatory system", "", module_funcset$Name)

# Plot
module_funcset_plot <- 
  ggplot(module_funcset, aes(x = reorder(Name, cor_value), y = cor_value, fill = Small_category)) +
  geom_bar(stat = "identity") +
  coord_flip() +
  labs(y = "Spearman correlation", x = "", title = "") +
  theme_classic() +
  theme(axis.title = element_text(size = 12),
        axis.text = element_text(size = 12),
        legend.text = element_text(size = 12),
        legend.position = "bottom", 
        legend.title = element_blank()) +
  scale_y_continuous(limits = c(-1, 1)) + # Adjust the y-axis scale
  scale_fill_brewer(palette = "Set3")

print(module_pathway_complex_plot)
print(module_funcset_plot)

# Save
ggsave(paste0(Figure_path, "Figure5A_module_spearman.pdf"), module_pathway_complex_plot, width = 8, height = 6, dpi = 300)
ggsave(paste0(Figure_path, "Figure5B_module_spearman.pdf"), module_funcset_plot, width = 8, height = 6, dpi = 300)

```


